modules/cli/src/main/scala/scala/cli/commands/InstallHome.scala

```diff
@@ -4,64 +4,102 @@ import caseapp._
+      s"Scala-cli $version is already installed and up-to-date."
+    )
+    sys.exit(0)
+  }
+
+  private def loggingUpdate(env: Boolean, newVersion: String, oldVersion: String) = {
+    if (!env) println(
+      s"""scala-cli $oldVersion is already installed and out-of-date.
+         |scala-cli will be updated to version $newVersion
+         |""".stripMargin
+    )
+  }
+
+  private def loggingDowngrade(env: Boolean, newVersion: String, oldVersion: String) = {
+    if (!env && coursier.paths.Util.useAnsiOutput()) {
+      println(s"scala-cli $oldVersion is already installed and up-to-date.")
+      println(s"Do you want to downgrade scala-cli to version $newVersion [Y/n]")
+      val response = readLine()
+      if (response != "Y") {
+        System.err.println("Abort")
         sys.exit(1)
       }
     }
+    else {
+      System.err.println(
+        s"Error: scala-cli is already installed $oldVersion and up-to-date. Downgrade to $newVersion pass -f or --force."
+      )
+      sys.exit(1)
+    }
+  }
+
+  def run(options: InstallHomeOptions, args: RemainingArgs): Unit = {
+
+    val binDirPath =
+      options.binDirPath.getOrElse(scala.build.Directories.default().binRepoDir / "scala-cli")
+    val newScalaCliBinPath = os.Path(options.scalaCliBinaryPath, os.pwd)
+
+    val newVersion: String =
+      os.proc(newScalaCliBinPath, "version").call(cwd = os.pwd).out.text.trim
+
+    // Backward compatibility - previous versions not have the `--version` parameter
+    val oldVersion: String = Try {
+      os.proc(binDirPath / options.binaryName, "version").call(cwd = os.pwd).out.text.trim
+    }.toOption.getOrElse("0.0.0")
+
+    if (os.exists(binDirPath)) {
+      if (options.force) () // skip logging
+      else if (newVersion == oldVersion) loggingEqual(newVersion)
+      else if (isOutOfDate(newVersion, oldVersion))
+        loggingUpdate(options.env, newVersion, oldVersion)
+      else loggingDowngrade(options.env, newVersion, oldVersion)
+    }
+
+    os.remove.all(binDirPath)
 
     os.copy(
-      from = os.Path(options.scalaCliBinaryPath, os.pwd),
-      to = scalaCliBinPath / "scala-cli",
+      from = newScalaCliBinPath,
+      to = binDirPath / options.binaryName,
       createFolders = true
     )
     if (!Properties.isWin)
-      os.perms.set(scalaCliBinPath / "scala-cli", os.PermSet.fromString("rwxrwxr-x"))
-
-    val update = EnvironmentUpdate(Nil, Seq("PATH" -> scalaCliBinPath.toString()))
-
-    val didUpdate =
-      if (Properties.isWin) {
-        val updater = CustomWindowsEnvVarUpdater().withUseJni(Some(coursier.paths.Util.useJni()))
-        updater.applyUpdate(update)
-      }
-      else {
-        val updater = ProfileUpdater()
-        updater.applyUpdate(update)
-      }
+      os.perms.set(binDirPath / options.binaryName, os.PermSet.fromString("rwxr-xr-x"))
 
-    if (didUpdate) {
-      println("Successfully installed scala-cli")
+    if (options.env) {
+      println(s""" export PATH="$$PATH:$binDirPath" """)
```

Comment:
Don't add spaces around the command (NIT), and maybe put the directory first here (it's meant to apply only to the current session, so it should be safe to put our directory first here)
```suggestion
      println(s"""export PATH="$binDirPath:$$PATH"""")
```